### PR TITLE
[ESQL] Refactor Element Type selection to use a switch

### PR DESCRIPTION
**Pull request by not-napoleon**
Similar to https://github.com/elastic/elasticsearch/pull/110036, this creates the opportunity for the compiler to flag this as a place which needs to be changed when we add a data type.  It's also fewer lines of code, and makes it explicit which types are not supported.

Ideally, this would also become a property on DataType, but that can't happen until we pull it out of core.

### CODE REVIEW

**Review by nik9000 — APPROVED** [review]

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/PlannerUtils.java [hunk 1]
```diff
@@ -296,6 +272,6 @@ public static ElementType toElementType(DataType dataType, MappedFieldType.Field
      * Returns DOC_VALUES if the given boolean is set.
      */
     public static MappedFieldType.FieldExtractPreference extractPreference(boolean hasPreference) {
-        return hasPreference ? MappedFieldType.FieldExtractPreference.DOC_VALUES : MappedFieldType.FieldExtractPreference.NONE;
+        return hasPreference ? DOC_VALUES : MappedFieldType.FieldExtractPreference.NONE;
```
Comment (bpintea) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/PlannerUtils.java:275]: Nit: might be worth statically importing `MappedFieldType.FieldExtractPreference.NONE` as well, they're part of the same enum.

**Review by bpintea — APPROVED** [review]
LGTM

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/PlannerUtils.java [hunk 2]
```diff
@@ -239,47 +240,23 @@ public static ElementType toElementType(DataType dataType) {
...
+            case INTEGER, COUNTER_INTEGER -> ElementType.INT;
+            case DOUBLE, COUNTER_DOUBLE -> ElementType.DOUBLE;
+            // unsupported fields are passed through as a BytesRef
+            case KEYWORD, TEXT, IP, SOURCE, VERSION, UNSUPPORTED -> ElementType.BYTES_REF;
+            case NULL -> ElementType.NULL;
+            case BOOLEAN -> ElementType.BOOLEAN;
+            case DOC_DATA_TYPE -> ElementType.DOC;
+            case TSID_DATA_TYPE -> ElementType.BYTES_REF;
+            case GEO_POINT, CARTESIAN_POINT -> fieldExtractPreference == DOC_VALUES ? ElementType.LONG : ElementType.BYTES_REF;
             // TODO: support forStats for shape aggregations, like st_centroid
```
Comment (craigtaverner) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/PlannerUtils.java:255]: This comment is out of date, and can be removed. When we do doc-values they will almost certainly also be backed by ByteRefs, so no consideration for fieldExtractPreference is needed (the term `forStats` was used before we moved to FieldExtractPreference).

Comment (craigtaverner) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/PlannerUtils.java:255]:
```suggestion
```

**Comment by not-napoleon** [comment]
@elasticmachine run elasticsearch-ci/part-3

**Comment by not-napoleon** [comment]
Test failure reproduces on main, so it's not related to this PR.  I opened an issue for it: https://github.com/elastic/elasticsearch/issues/110152

**Pull request merged**